wire-ios-sync-engine/Source/Repositories/TeamRepository.swift

```diff
@@ -0,0 +1,261 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireDataModel
+
+/// Facilitate access to team related domain objects.
+///
+/// A repository provides an abstraction for the access and storage
+/// of domain models, concealing how and where the models are stored
+/// as well as the possible source(s) of the models.
+
+protocol TeamRepositoryProtocol {
+
+    /// Pull self team metadata frmo the server and store locally.
+
+    func pullSelfTeam() async throws
+
+    /// Pull team roles for the self team from the server and store locally.
+
+    func pullSelfTeamRoles() async throws
+
+    /// Pull team members for the self team from the server and store locally.
+
+    func pullSelfTeamMembers() async throws
+
+    /// Fetch the legalhold status for the self user from the server.
+
+    func fetchSelfLegalholdStatus() async throws -> LegalholdStatus
```

Comment:
Do I understand correctly, that `pull` should describe also the locale store? I like the idea

Reply:
I'm not sure what you mean by "describe also the locale store".